presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -126,21 +128,19 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
                     .append(switchCaseBlocks);
         }
         else {
-            // TODO: replace Set with fastutils (or similar) primitive sets if types are primitive
-            // for huge IN lists, use a Set
-            Binding constant = generatorContext.getCallSiteBinder().bind(constantValues, Set.class);
+            // for huge IN lists, use a Fastutil primitive Set
+            Set<?> fastutilConstantValuesSet = toFastutilHashSet(constantValues, javaType);
+            Binding constant = generatorContext.getCallSiteBinder().bind(fastutilConstantValuesSet, fastutilConstantValuesSet.getClass());
 
             switchBlock = new Block()
                     .comment("inListSet.contains(<stackValue>)")
                     .append(new IfStatement()
                             .condition(new Block()
-                                    .comment("value (+boxing if necessary)")
+                                    .comment("value")
                                     .dup(javaType)
-                                    .append(ByteCodeUtils.boxPrimitive(javaType))
                                     .comment("set")
                                     .append(loadConstant(constant))
-                                    // TODO: use invokeVirtual on the set instead. This requires swapping the two elements in the stack
-                                    .invokeStatic(CompilerOperations.class, "in", boolean.class, Object.class, Set.class))
+                                    .invokeStatic(FastutilSetHelper.class, "in", boolean.class, javaType.isPrimitive() ? javaType : Object.class, fastutilConstantValuesSet.getClass()))
```

Comment:
Pull the class expression out into a variable and put it above, after the set is created, since it is a bit long